### DISCUSSION TITLE
Reutilize InflaterInputStream

### DISCUSSION THREAD

**Discussion by rickomax** [icsharpcode/SharpZipLib#548] [Q&A]
Hi!
Not really a bug but more a question.

I'm using InflaterInputStream to inflate some data, and each data is located at different places of my original MemoryStream.
I'm trying to avoid creating many InflaterInputStream instances as I want to keep buffer allocations low. I have created a single InflaterInputStream instance to do that.  After the first data instance is inflated successfully, the other ones can't be inflated anymore since the OutputWindow isn't filled anymore.

Is there a way to handle that, or I really need to create a new InflaterInputStream instance for every data instance I want to inflate?

**Comment by piksel** [comment]
There is a `Fill()` method that seems to do that, but I think the only "official" way is to create a new instance every time you want to restart the inflation. Another alternative would of course be to manually use the `Inflator` instead, but it's a lot more overhead to get it working probably.

**Reply by rickomax** [reply]
Well, I guess the best way to do that would be using the Inflator instead. That will require studying all the process in-depth. I've tried calling the Fill method manually by overriding the InflaterInputStream class, but without success.

**Reply by rickomax** [reply]
Calling Inflater.Reset before every read works until a certain point.

**Comment by rickomax** [comment]
I did some tests, and it looks like there is no way to reutilize even the Inflater since it has some internal variables that will get messed up when using the same instance and multiple Enumerators (which is my case).
I did the best I could by using an Inflater pool and reutilize the Inflaters as much as possible.
